Homeworks/Homework_5/transfers_script.py

```diff
@@ -0,0 +1,347 @@
+    :param money_amount: (float) The amount of money to be transferred.
+    :return str: The status of the operation.
+    """
+    # PREPARE THE DATA FOR FURTHER USE
+    cursor.execute(SELECT_COMMAND.format('Amount, Currency', 'Account', sender_id))
+    sender_amount, sender_currency = cursor.fetchone()
+
+    # TERMINATE THE EXECUTION IN CASE OF LACK OF FUNDS
+    if sender_amount < money_amount:
+        return 'The sender does not have enough money'
+
+    # PREPARE THE DATA FOR FURTHER USE
+    cursor.execute(SELECT_COMMAND.format('Currency', 'Account', recipient_id))
+    recipient_currency = cursor.fetchone()[0]
+
+    currency = get_currency()
+
+    cr = 1 if sender_currency == recipient_currency else currency[sender_currency]
+    # CONVERT CURRENCY
+    converted_amount = money_amount * cr
+
+    # MAKE THE TRANSFER
+    return make_approved_transfer(cursor, money_amount, converted_amount, recipient_id, sender_id, sender_currency)
+
+
+def make_approved_transfer(cursor, money_amount, converted_amount, recipient_id, sender_id, sender_currency):
+    """
+    Execute an approved money transfer between accounts and record the transaction.
+
+    :param cursor: (sqlite3.Cursor) The cursor to interact with the database.
+    :param money_amount: (float) The original amount of money to be transferred.
+    :param converted_amount: (float) The amount of money after currency conversion.
+    :param recipient_id: (int) The ID of the recipient's account.
+    :param sender_id: (int) The ID of the sender's account.
+    :param sender_currency: (str) The currency of the sender's account.
+
+    :return: (str) A status message indicating the successful completion of the transfer.
+
+    :raises: Any database-related exceptions that may occur during execution.
+    """
+    # RECIPIENT
+    cursor.execute(UPDATE_COMMAND.format('+', converted_amount, recipient_id))
+    # SENDER
+    cursor.execute(UPDATE_COMMAND.format('-', money_amount, sender_id))
+
+    # FILL TRANSACTIONS TABLE
+    cursor.execute(SELECT_COMMAND.format('Bank_id', 'Account', sender_id))
+    id_bank_sender = cursor.fetchone()[0]
+    cursor.execute(SELECT_COMMAND.format('Bank_id', 'Account', recipient_id))
+    id_bank_receiver = cursor.fetchone()[0]
+
+    cursor.execute(SELECT_COMMAND.format('name', 'Bank', id_bank_sender))
+    bank_sender_name = cursor.fetchone()[0]
+    cursor.execute(SELECT_COMMAND.format('name', 'Bank', id_bank_receiver))
+    bank_receiver_name = cursor.fetchone()[0]
+
+    cursor.execute('INSERT INTO Transactions (Bank_sender_name, Account_sender_id, '
+                   'Bank_receiver_name, Account_receiver_id, Sent_Currency, Sent_Amount, Datetime) '
+                   'VALUES (?, ?, ?, ?, ?, ?, ?)',
+                   (bank_sender_name, sender_id, bank_receiver_name, recipient_id,
+                    sender_currency, money_amount,
+                    datetime.now(timezone.utc).strftime('%Y-%m-%d %H:%M:%S')))
```

Comment:
![image](https://github.com/MykytaKa/PyCourse/assets/43804299/f8f381ff-e891-4ab9-9dbf-06336e814615)
but you even don't provide ability to pass it